Tests/Runtime/Session/BacktraceSessionTests.cs

```diff
@@ -0,0 +1,282 @@
+            var backtraceSession = new BacktraceSession(_attributeProvider, _submissionUrl, timeInterval, 3);
+            var requestHandler = new BacktraceHttpClientMock();
+            backtraceSession.RequestHandler = requestHandler;
+
+            backtraceSession.AddSessionEvent(SessionEventName);
+            backtraceSession.AddUniqueEvent(UniqueAttributeName);
+
+            backtraceSession.Tick(timeInterval - 1);
+
+            Assert.AreEqual(numberOfAddedEvents, backtraceSession.Count());
+            Assert.AreEqual(0, requestHandler.NumberOfRequests);
+        }
+
+
+        [Test]
+        public void BacktraceSession_ShouldTriggerDownloadAfterTimeIntervalHitAgain_DataWasSendToBacktrace()
+        {
+            const int timeInterval = 10;
+            var backtraceSession = new BacktraceSession(_attributeProvider, _submissionUrl, timeInterval, 3);
+            var requestHandler = new BacktraceHttpClientMock();
+            backtraceSession.RequestHandler = requestHandler;
+
+            backtraceSession.AddSessionEvent(SessionEventName);
+            backtraceSession.AddUniqueEvent(UniqueAttributeName);
+
+            backtraceSession.Tick(timeInterval + 1);
+
+            backtraceSession.AddSessionEvent(SessionEventName);
+            backtraceSession.AddUniqueEvent(UniqueAttributeName);
+
+            backtraceSession.Tick((timeInterval * 2) + 1);
+
+            Assert.AreEqual(0, backtraceSession.Count());
+            Assert.AreEqual(2, requestHandler.NumberOfRequests);
+        }
+        [Test]
+        public void BacktraceSession_ShouldntTriggerDownloadAfterTimeIntervalFirstHit_DataWasSendToBacktrace()
+        {
+            const int timeInterval = 10;
+            var backtraceSession = new BacktraceSession(_attributeProvider, _submissionUrl, timeInterval, 3);
+            var requestHandler = new BacktraceHttpClientMock();
+            backtraceSession.RequestHandler = requestHandler;
+
+            backtraceSession.AddSessionEvent(SessionEventName);
+            backtraceSession.AddUniqueEvent(UniqueAttributeName);
+
+            backtraceSession.Tick(timeInterval + 1);
+
+            backtraceSession.AddSessionEvent(SessionEventName);
+            backtraceSession.AddUniqueEvent(UniqueAttributeName);
+
+            backtraceSession.Tick(timeInterval + 2);
+
+            Assert.AreEqual(2, backtraceSession.Count());
+            Assert.AreEqual(1, requestHandler.NumberOfRequests);
+        }
+
+        [Test]
+        public void BacktraceSessionDefaultEvent_ShouldSendDefaultEventOnTheApplicationStartup_DataWasSendToBacktrace()
+        {
+            var backtraceSession = new BacktraceSession(_attributeProvider, _submissionUrl, 10, 3);
+            var requestHandler = new BacktraceHttpClientMock();
+            backtraceSession.RequestHandler = requestHandler;
+
+            backtraceSession.SendStartupEvent();
+
+            Assert.AreEqual(1, requestHandler.NumberOfRequests);

```

Comment:
Should we validate any `StartupEvent` params?

Reply:
It's constant and unfortunately this function won't store anything on the event queue - it will create in memory a json object and send it : (